lib/puppet/type/chocolateyconfig.rb

```diff
@@ -0,0 +1,77 @@
+require 'puppet/type'
+require 'pathname'
+
+Puppet::Type.newtype(:chocolateyconfig) do
+
+  @doc = <<-'EOT'
+    Allows managing config settings for Chocolatey.
+    Configuration values provide settings for users
+    to configure aspects of Chocolatey and the way it
+    functions. Similar to features, except allow for user
+    configured values. Learn more about config at
+    https://chocolatey.org/docs/commands-config
+  EOT
+
+  ensurable do
+    newvalue(:present)  { provider.create }
+    newvalue(:absent)   { provider.destroy }
+    defaultto :present
+
+    def retrieve
+      provider.properties[:ensure]
+    end
+
+  end
+
+  newparam(:name) do
+    desc "The name of the config setting. Used for uniqueness.
+      Puppet is not able to easily manage any values that
+      include Password in the key name in them as they
+      will be encrypted in the configuration file."
+
+    validate do |value|
+      if value.nil? or value.empty?
+        raise ArgumentError, "A non-empty name must be specified."
+      end
+    end
+
+    isnamevar
+
+    munge do |value|
+      value.downcase
+    end
+
+    def insync?(is)
+      is.downcase == should.downcase
+    end
+  end
+
+  newproperty(:value) do
+    desc "The value of the config setting. If the
+      name includes Password, then the value is
+      not ensurable due to being encrypted in the
+      configuration file."
+
+    validate do |value|
+      if value.nil? or value.empty?
+        raise ArgumentError, "A non-empty value must be specified. To unset value, use ensure => absent"
+      end
+    end
+
+    def insync?(is)
+      #todo: If name contains password, it is
+      # always in sync if there is a value
```

Comment:
If the name contains password, we need to state that it is insync if there is a value.